src/h5web/visualizations/shared/AxisSystem.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { useState } from 'react';
+import { useThree, useFrame, Dom } from 'react-three-fiber';
+import { scaleLinear } from 'd3-scale';
+import { AxisLeft, AxisBottom } from '@vx/axis';
+import { format } from 'd3-format';
+import { TickRendererProps } from '@vx/axis/lib/types';
+import { Grid } from '@vx/grid';
+import styles from './AxisSystem.module.css';
+import { Domain, AxisOffsets, AxisDomains } from './models';
+import { adaptedNumTicks } from './utils';
+
+interface Props {
+  axisDomains: AxisDomains;
+  axisOffsets: AxisOffsets;
+  showGrid?: boolean;
+}
+
+function AxisSystem(props: Props): JSX.Element {
+  const { axisDomains, axisOffsets, showGrid } = props;
+
+  const { camera, size } = useThree();
+  const { width, height } = size;
+
+  const [domains, setDomains] = useState<AxisDomains>(axisDomains);
+
+  // Axis bounds in R3F camera coordinates
+  const leftAxisBounds = [-height / 2, height / 2];
+  const bottomAxisBounds = [-width / 2, width / 2];
+
+  // Scales R3F camera coordinates to axis bounds
+  const leftAxisScale = scaleLinear()
+    .domain(leftAxisBounds)
+    .range(axisDomains.left);
+  const bottomAxisScale = scaleLinear()
```

Comment:
Maybe this needs renaming now. These scales are used to compute the bounds/domain of the axis as opposed to the ticks inside the axis.